ktor-io/posix/src/io/ktor/utils/io/concurrent/SharedNative.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2014-2020 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.utils.io.concurrent
+
+import io.ktor.utils.io.core.internal.*
+import kotlinx.atomicfu.*
+import kotlin.native.concurrent.*
+import kotlin.properties.*
+import kotlin.reflect.*
+
+/**
+ * Allows to create mutate property with frozen value.
+ *
```

Comment:
Please, note that if someone passes a `class X(var y: Int)` as `T` it will be frozen in Kotlin/Native due to the implementation of AtomicReference there. Hence, you can mutate `x` in this case but you can not mutate `x.y`.